### PR TITLE
accept connection string in CosmosClient constructor

### PR DESCRIPTION
**Pull request by jdanyow**

### CODE REVIEW

File: .changeset/cuddly-bottles-smell.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@cfworker/cosmos': patch
```
Comment (ItalyPaleAle) [inline .changeset/cuddly-bottles-smell.md:2]: Should this be marked as major? It looks like it'll be a big breaking change

**Comment by ItalyPaleAle** [comment]
While I understand the motive behind renaming the config option and I don't disagree with that in principle, I'd like to point out that Cosmos DB does use the expression "master key" in products and docs (https://docs.microsoft.com/en-us/rest/api/cosmos-db/access-control-on-cosmosdb-resources#master-key-tokens). This seems to be a major breaking change that would also use different names from the product.

It's your choice ultimately, but if you do decide to move forward with the renaming please consider bumping the major version per semver and avoid breaking all of our projects :)

**Comment by jdanyow** [comment]
I shipped a minor (not patch) release- safe to ignore the changeset, it wasn't used as written. Probably should have been a major bump.

I had always been on the fence with regards to the masterKey naming because future releases might support master key tokens *and resource tokens*. Your PR pushed me over the edge due to the addition of AccountKey. Maybe it should be `token`... thoughts? Easy to ship another release with major version bump, I'd rather get this right.

**Comment by ItalyPaleAle** [comment]
1. I would suggest unpublishing 1.13 from NPM. By default NPM adds packages as `^1.12.0` which means that on the next `npm install` users will get 1.13 automatically. Even if 2.0 comes out, users who have pinned `^1.12.0` will still get 1.13. That'll break a lot of apps.
2. I totally agree that "master key" is ill-named, but sadly that's how it's called in the product. If you do want to migrate to "account key" (I personally wouldn't because it'd be inconsistent with the name used in the product, but it's your choice as maintainer of this project), I really think that should have been a major version bump.
3. In any case, please note #203. In the connection string it **must** be called `MasterKey` because that's how connection strings returned by the service (such as in the Azure Portal) are formatted.

**Pull request merged**